src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,45 @@
+package tasks;
+
+/**
+ * Represents a general Task object with a name.
+ */
+
+public abstract class Task {
+    private String name;
+    private boolean isDone;
+
+    /**
+     * Creates a new Task object.
+     * @param name Name of the task.
+     * @param isDone Whether the task is done or not.
+     */
+    public Task(String name, boolean isDone) {
+        this.name = name;
+        this.isDone = isDone;
+    }
+
+    public void setDone(boolean isDone) {
+        this.isDone = isDone;
+    }
+
+    /**
+     * Prints a tick or X symbol.
+     * @return a tick or X symbol.
+     */
+
```

Comment:
Maybe can remove the enter spacing here so it doesn't seem like a separate method javadoc